locale.c

```diff
@@ -2854,13 +2865,25 @@ S_calculate_LC_ALL_string(pTHX_ const char ** category_locales_list,
                 retval = savepv(retval);
                 SAVEFREEPV(retval);
             }
+        }
 
         /* In all cases here, there's nothing we create that needs to be freed,
          * so leave 'free_if_void_return' set to the default 'false'. */
     }
     else {  /* Here, not all categories have the same locale */
-        char * writable_alias;  /* XXX This name and indentation will make
-                                   sense in a few commits */
+        char * writable_alias;
+
+        /* If returning to PL_setlocale_buf, set up to write directly to it,
+         * being sure it is resized to be large enough */
+        if (returning == WANT_PL_setlocale_buf) {
+            if (total_len > PL_setlocale_bufsize) {
+                PL_setlocale_bufsize = total_len;
+                Renew(PL_setlocale_buf, PL_setlocale_bufsize, char);
+            }
+
+            writable_alias = (char *) PL_setlocale_buf;
```

Comment:
I think the bug here is that `PL_setlocale_buf` is `const  char *`.

save_to_buffer() gets away with no warning because Copy() brokenly casts away the const.

Reply:
I've been looking at the trees here and not the forest.  This doesn't lead to incorrect results.  The goal is to be able to return in a variable that XS code shouldn't bet able to write to, but we have to be able to initialize and manipulate the string before returning.  What do you suggest?

Reply:
They can modify it in the same way you are, by using Copy(), or assigning a newly allocated pointer.

Or by casting as you do.

You can't protect it, I think we're better off not getting any more const unsafe.